src/code/audio_heap.c

```diff
@@ -1,106 +1,1386 @@
+                            break;
+                        case 1:
+                            Audio_DiscardBank(tp->entries[0].id);
+                            break;
+                    }
+
+                    tp->entries[0].id = -1;
+                    pool->cur = pool->start;
+                }
+                ret = tp->entries[1].ptr;
+                break;
+
+            default:
+                return NULL;
+        }
+
+        tp->nextSide ^= 1;
+        return ret;
+    }
+
+    mem = Audio_Alloc(&loadedPool->persistent.pool, size);
+    loadedPool->persistent.entries[loadedPool->persistent.numEntries].ptr = mem;
+
+    if (mem == NULL) {
+        switch (arg2) {
+            case 2:
+                return Audio_AllocBankOrSeq(poolIdx, size, 0, id);
+
+            case 0:
+            case 1:
+                return NULL;
+        }
+    }
+
+    loadedPool->persistent.entries[loadedPool->persistent.numEntries].id = id;
+    loadedPool->persistent.entries[loadedPool->persistent.numEntries].size = size;
+    return loadedPool->persistent.entries[loadedPool->persistent.numEntries++].ptr;
+}
+
+void* func_800DF074(s32 poolIdx, s32 arg1, s32 id) {
+    void* ret;
+
+    ret = func_800E04E8(poolIdx, id);
+    if (ret != NULL) {
+        return ret;
+    }
+    if (arg1 == 3) {
+        return NULL;
+    }
+    return func_800DF0CC(poolIdx, arg1, id);
+}
+
+void* func_800DF0CC(s32 poolIdx, s32 arg1, s32 bankId) {
+    u32 i;
+    SoundMultiPool* loadedPool;
+    TemporaryPool* temporary;
+    PersistentPool* persistent;
+
+    switch (poolIdx) {
+        case 0:
+            loadedPool = &gAudioContext.gSeqLoadedPool;
+            break;
+        case 1:
+            loadedPool = &gAudioContext.gBankLoadedPool;
+            break;
+        case 2:
+            loadedPool = &gAudioContext.gUnusedLoadedPool;
+            break;
+    }
+
+    temporary = &loadedPool->temporary;
+    if (arg1 == 0) {
+        if (temporary->entries[0].id == bankId) {
+            temporary->nextSide = 1;
+            return temporary->entries[0].ptr;
+        } else if (temporary->entries[1].id == bankId) {
+            temporary->nextSide = 0;
+            return temporary->entries[1].ptr;
+        } else {
+            return NULL;
+        }
+    }
+
+    persistent = &loadedPool->persistent;
+    for (i = 0; i < persistent->numEntries; i++) {
+        if (persistent->entries[i].id == bankId) {
+            return persistent->entries[i].ptr;
+        }
+    }
+
+    if (arg1 == 2) {
+        return func_800DF074(poolIdx, 0, bankId);
+    }
+    return NULL;
+}
+
+void func_800DF1D8(f32 arg0, f32 arg1, u16* arg2) {
+    s32 i;
+    f32 tmp[16];
+
+    tmp[0] = (f32)(arg1 * 262159.0f);
+    tmp[8] = (f32)(arg0 * 262159.0f);
+    tmp[1] = (f32)((arg1 * arg0) * 262159.0f);
+    tmp[9] = (f32)(((arg0 * arg0) + arg1) * 262159.0f);
+
+    for (i = 2; i < 8; i++) {
+        //! @bug they probably meant to store the value to tmp[i] and tmp[8 + i]
```

Comment:
code looks good to me, only thing i would change is we try to stay away from referring to the devs directly. would phrase it as like "value was probably meant to be stored to tmp[i] and tmp[8 + i]"